src/rust/ide/enso-protocol/src/file_manager.rs

```diff
@@ -0,0 +1,411 @@
+//! Client library for the JSON-RPC-based File Manager service.
+
+#![warn(missing_docs)]
+#![warn(trivial_casts)]
+#![warn(trivial_numeric_casts)]
+#![warn(unused_import_braces)]
+#![warn(unused_qualifications)]
+#![warn(unsafe_code)]
+#![warn(missing_copy_implementations)]
+#![warn(missing_debug_implementations)]
+
+use crate::prelude::*;
+
+use crate::common::UTCDateTime;
+
+use json_rpc::api::Result;
+use json_rpc::Handler;
+use json_rpc::make_arg;
+use json_rpc::make_param_map;
+use json_rpc::make_rpc_methods;
+use futures::Stream;
+use serde::Serialize;
+use serde::Deserialize;
+use std::future::Future;
+use uuid::Uuid;
+
+
+
+// =============
+// === Event ===
+// =============
+
+/// Event emitted by the File Manager `Client`.
+pub type Event = json_rpc::handler::Event<Notification>;
+
+
+
+// ============
+// === Path ===
+// ============
+
+/// Path to a file.
+#[derive(Clone,Debug,Display,Eq,Hash,PartialEq,PartialOrd,Ord)]
+#[derive(Serialize, Deserialize)]
+#[derive(Shrinkwrap)]
+pub struct Path(pub String);
```

Comment:
Please do not make `.0` field public. Whatit means if I access a `.0` field on path? This is hacky and show old never be a public API?

Reply:
I think this is mainly for being able to create path `Path(some_string)`

Reply:
@farmaazon Then create `fn Path(...)` and silence warning about fn with upper case letter - this will give you both nice constructor and lack of ugly `.0` API :)

Reply:
@wdanilo I will leave it as it is. Path is being reworked in the File Manager Client task.

Reply:
Then create fixme in the code about it.